components/registration-framework/org.wso2.carbon.identity.user.registration.engine/src/main/java/org/wso2/carbon/identity/user/registration/engine/graph/TaskExecutionNode.java

```diff
@@ -0,0 +1,148 @@
+import static org.wso2.carbon.identity.user.registration.engine.util.Constants.ExecutorStatus.STATUS_USER_CREATED;
+import static org.wso2.carbon.identity.user.registration.engine.util.Constants.ExecutorStatus.STATUS_USER_INPUT_REQUIRED;
+import static org.wso2.carbon.identity.user.registration.engine.util.Constants.STATUS_COMPLETE;
+import static org.wso2.carbon.identity.user.registration.engine.util.Constants.STATUS_INCOMPLETE;
+import static org.wso2.carbon.identity.user.registration.engine.util.RegistrationFlowEngineUtils.handleServerException;
+import static org.wso2.carbon.identity.user.registration.mgt.Constants.NodeTypes.TASK_EXECUTION;
+import static org.wso2.carbon.identity.user.registration.mgt.Constants.StepTypes.REDIRECTION;
+import static org.wso2.carbon.identity.user.registration.mgt.Constants.StepTypes.VIEW;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.user.registration.engine.internal.RegistrationFlowEngineDataHolder;
+import org.wso2.carbon.identity.user.registration.engine.model.ExecutorResponse;
+import org.wso2.carbon.identity.user.registration.engine.model.RegisteringUser;
+import org.wso2.carbon.identity.user.registration.engine.model.RegistrationContext;
+import org.wso2.carbon.identity.user.registration.engine.model.Response;
+import org.wso2.carbon.identity.user.registration.mgt.exception.RegistrationFrameworkException;
+import org.wso2.carbon.identity.user.registration.mgt.exception.RegistrationServerException;
+import org.wso2.carbon.identity.user.registration.mgt.model.NodeConfig;
+
+/**
+ * Implementation of a node specific to executing a registration executor.
+ */
+public class TaskExecutionNode implements Node {
+
+    private static final Log LOG = LogFactory.getLog(TaskExecutionNode.class);
+
+    @Override
+    public String getName() {
+
+        return TASK_EXECUTION;
+    }
+
+    @Override
+    public Response execute(RegistrationContext context, NodeConfig configs)
+            throws RegistrationFrameworkException {
+
+        if (configs.getExecutorConfig() == null) {
+            throw handleServerException(ERROR_CODE_EXECUTOR_NOT_FOUND, context.getRegGraph().getId(),
+                                        context.getTenantDomain());
+        }
+
+        Executor mappedRegExecutor = resolveExecutor(configs, context.getRegGraph().getId(), context.getTenantDomain());
+        // TODO: 2021/4/6 Add executor configuration to the context where application
+        return triggerExecutor(context, mappedRegExecutor);
+    }
+
+    private Executor resolveExecutor(NodeConfig configs, String graphId, String tenantDomain)
+            throws RegistrationServerException {
+
+        String executorName = configs.getExecutorConfig().getName();
+        Executor mappedRegExecutor = null;
+        for (Executor executor : RegistrationFlowEngineDataHolder.getInstance().getExecutors()) {
```

Comment:
As we are fetching the executor by name, if we kept the executors in a hashmap (executor against the name) it would avoid iterating through all executors.